systemtest/src/test/java/io/strimzi/systemtest/dynamicconfiguration/DynamicConfigurationIsolatedST.java

```diff
@@ -0,0 +1,374 @@
+                                .withTls(false)
+                            .endKafkaListenerExternalLoadBalancer()
+                        .endListeners()
+                        .withConfig(kafkaConfig)
+                    .endKafka()
+                .endSpec()
+                .done();
+
+        KafkaTopicResource.topic(CLUSTER_NAME, TOPIC_NAME).done();
+        KafkaUserResource.tlsUser(CLUSTER_NAME, USER_NAME).done();
+
+        BasicExternalKafkaClient basicExternalKafkaClientTls = new BasicExternalKafkaClient.Builder()
+            .withTopicName(TOPIC_NAME)
+            .withNamespaceName(NAMESPACE)
+            .withClusterName(CLUSTER_NAME)
+            .withMessageCount(MESSAGE_COUNT)
+            .withKafkaUsername(USER_NAME)
+            .withConsumerGroupName(CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE))
+            .withSecurityProtocol(SecurityProtocol.SSL)
+            .build();
+
+        BasicExternalKafkaClient basicExternalKafkaClientPlain = new BasicExternalKafkaClient.Builder()
+            .withTopicName(TOPIC_NAME)
+            .withNamespaceName(NAMESPACE)
+            .withClusterName(CLUSTER_NAME)
+            .withMessageCount(MESSAGE_COUNT)
+            .withConsumerGroupName(CONSUMER_GROUP_NAME + "-" + rng.nextInt(Integer.MAX_VALUE))
+            .withSecurityProtocol(SecurityProtocol.PLAINTEXT)
+            .build();
+
+        String userName = "john";
+        KafkaUserResource.tlsUser(CLUSTER_NAME, userName).done();
+
+        basicExternalKafkaClientTls.setKafkaUsername(userName);
+
+        basicExternalKafkaClientPlain.verifyProducedAndConsumedMessages(
+                basicExternalKafkaClientPlain.sendMessagesPlain(),
+                basicExternalKafkaClientPlain.receiveMessagesPlain()
+        );
+
+        assertThrows(Exception.class, () -> {
+            basicExternalKafkaClientTls.sendMessagesTls(Constants.GLOBAL_CLIENTS_EXCEPT_ERROR_TIMEOUT);
+            basicExternalKafkaClientTls.receiveMessagesTls(Constants.GLOBAL_CLIENTS_EXCEPT_ERROR_TIMEOUT);
+            LOGGER.error("Producer & Consumer did not send and receive messages because external listener is set to plain communication");
+        });
+
+        LOGGER.info("Updating listeners of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaListeners updatedKl = new KafkaListenersBuilder()
+                    .withNewKafkaListenerExternalNodePort()
+                        .withNewKafkaListenerAuthenticationTlsAuth()
+                        .endKafkaListenerAuthenticationTlsAuth()
+                    .endKafkaListenerExternalNodePort()
+                    .build();
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            kafkaClusterSpec.setListeners(updatedKl);
+        });
+
+        PodUtils.verifyThatRunningPodsAreStable(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME));
```

Comment:
Are you saying that the change here doesn't trigger RU?

Reply:
This should trigger a rolling update! Using method `verifyThatRunningPodsAreStable` is not a good practice here (again I did write these tests :D) and I have replaced it with the `snapShot` `waitTillSsHasRolled`.